### PR TITLE
Consolidate docs generation, improve noscript support

### PR DESCRIPTION
**Pull request by ras0219-msft**
This PR eliminates powershell as part of the docs generation script. The fundamental md -> html library we are using (`showdown`) is javascript, so this moves us closer to only requiring node.js in our infrastructure scripts.

Additionally:
1. The 'Source X Pro' webfont links have been removed because they violate the external web resources guidelines. If users have the fonts on their machines, they will still see those fonts, however other users will instead see more standard fonts (Segoe UI / Segoe Code).
2. All other resource references have had SRI added to conform to the external web resources guidelines. To fully conform, we also need to add locally hosted fallbacks but I'm leaving that for a separate PR.
3. Instead of having a prelude coming from a text file and generating a footer in the script, the docs pages are now generated from a mustache.js template. This also makes it extremely easy to directly embed the nav header and footer into the pages so that they are faster to load and render correctly without javascript.
4. The nav bar now starts expanded and is collapsed via javascript. This again improves the accessibility of the page without javascript.
5. Specification links now go directly to the vcpkg GitHub instead of indirecting through an empty page.

### CODE REVIEW

File: dist/common.js [hunk 1]
```diff
@@ -89,7 +88,7 @@ $(document).ready(function(){
     var siteConsent = null;
     WcpConsent.init("en-US", "banner", function (err, _siteConsent) {
         if (err != undefined) {
-            throw "error";
+            throw err;
         } else {
             siteConsent = _siteConsent;
         }
```
Comment (BillyONeal) [inline dist/common.js:94]:
```suggestion
        } 
        
        siteConsent = _siteConsent;
```

File: index.html [hunk 2]
```diff
@@ -1,6 +1,7 @@
 <!DOCTYPE html>
 <html>
   <head>
+    <script src="https://ajax.googleapis.com/ajax/libs/jquery/3.6.0/jquery.min.js"></script>
```
Comment (BillyONeal) [inline index.html:4]: We can't use Google's copy of JQuery under SDL guidelines.  See https://liquid.microsoft.com/Web/Object/Read/MS.Security/Requirements/Microsoft.Security.SystemsADM.10103

**Review by BillyONeal — CHANGES_REQUESTED** [review]

File: index.html (hunk 2 above)
Comment (ras0219-msft) [reply index.html:4]: We are already using their copy on the other pages; I just copied off of them. I'll look at replacing all the references.

Comment (BillyONeal) [reply index.html:4]: :'(

**Review by BillyONeal — APPROVED** [review]
Changed to approve for not making the problem worse :'(

File: scripts/generateDocs.js [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+        html.push('">');
+        html.push('<li class="list-can-expand">');
+        if (filename in mapTable) {
+            html.push(mapTable[filename]);
+        } else {
+            const data = await fs.readFile(fullName, 'utf-8');
+            const lines = data.split(/\r?\n/);
+            const matches = lines[0].match(/^#+ +(.*)/);
+            if (matches) {
+                html.push(matches[1].replace(/`(.*)`/g, '<code>$1</code>'));
```
Comment (BillyONeal) [inline scripts/generateDocs.js:168]: I'm a bit nervous about this not handling escaped \`s correctly but I suppose this isn't the end of the world

File: scripts/generateDocs.js [hunk 4]
```diff
@@ -0,0 +1,306 @@
...
+    translate.setOption('encodeEmails', false);
+
+    var html = translate.makeHtml(data);
+
+    html = html
+        .replace(/&lt;\/code&gt;/g, "</code>")
+        .replace("ports&amp;lt;package&amp;gt;\\", "ports\\&lt;package&gt;\\")
+        .replace(/<br \/>/g, " ")
+        .replace(/Vcpkg/g, "vcpkg")
+        .replace('<a href="mailto:vcpkg@microsoft.com">vcpkg@microsoft.com</a>', '<a href="&#x6d;&#x61;&#105;&#108;&#x74;&#111;&#58;v&#99;&#x70;&#107;&#x67;&#64;&#109;&#x69;&#x63;&#114;&#x6f;&#x73;&#x6f;&#102;&#x74;&#x2e;&#x63;&#x6f;&#x6d;">&#118;&#x63;&#x70;&#x6b;&#103;&#x40;&#109;&#x69;&#x63;&#114;&#111;&#x73;&#111;&#102;&#x74;&#x2e;&#99;&#x6f;&#x6d;</a>')
```
Comment (BillyONeal) [inline scripts/generateDocs.js:238]: It's a bit amusing to try to hide this email alias while having the cleartext email alias in the generation script here 🤷‍♂️

**Review by BillyONeal — APPROVED** [review]

**Pull request merged**